SimGeneral/CaloAnalysis/plugins/CaloTruthAccumulator.cc

```diff
@@ -1,461 +1,673 @@
+
   // now fill the calo particles
-  for( unsigned i = 0; i < output_.pCaloParticles->size(); ++i ) {
+  for (unsigned i = 0; i < output_.pCaloParticles->size(); ++i) {
     auto& cp = (*output_.pCaloParticles)[i];
-    for( unsigned j = m_caloParticles.sc_start_[i]; j < m_caloParticles.sc_stop_[i]; ++j ) {
-      edm::Ref<SimClusterCollection> ref(scHandle,j);
+    for (unsigned j = m_caloParticles.sc_start_[i]; j < m_caloParticles.sc_stop_[i]; ++j) {
+      edm::Ref<SimClusterCollection> ref(scHandle, j);
       cp.addSimCluster(ref);
     }
-  }  
+  }
 
-  event.put( std::move(output_.pCaloParticles), "MergedCaloTruth" );
+  event.put(std::move(output_.pCaloParticles), "MergedCaloTruth");
 
   calo_particles().swap(m_caloParticles);
 
-  std::unordered_map<Index_t,float>().swap(m_detIdToTotalSimEnergy);
-
-  std::unordered_map<Barcode_t,Index_t>().swap(m_genParticleBarcodeToIndex);
-  std::unordered_map<Barcode_t,Index_t>().swap(m_simTrackBarcodeToIndex);
-  std::unordered_map<Barcode_t,Index_t>().swap(m_genBarcodeToSimTrackIndex);
-  std::unordered_map<Barcode_t,Index_t>().swap(m_simVertexBarcodeToIndex);
-
-  std::unordered_multimap<Index_t,Index_t>().swap(m_detIdToCluster);
-  std::unordered_multimap<Barcode_t,Index_t>().swap(m_simHitBarcodeToIndex);
-  std::unordered_multimap<Barcode_t,Barcode_t>().swap(m_simVertexBarcodeToSimTrackBarcode);
-  std::unordered_map<Barcode_t,Barcode_t>().swap(m_simTrackBarcodeToSimVertexParentBarcode);
-  std::unordered_multimap<Barcode_t,Index_t>().swap(m_simTrackToSimVertex);
-  std::unordered_multimap<Barcode_t,Index_t>().swap(m_simVertexToSimTrackParent); 
-  std::vector<Barcode_t>().swap(m_simVertexBarcodes);
-  std::unordered_map<Index_t,float>().swap(m_detIdToTotalSimEnergy);
+  std::unordered_map<Index_t, float>().swap(m_detIdToTotalSimEnergy);
+  std::unordered_multimap<Barcode_t, Index_t>().swap(m_simHitBarcodeToIndex);
 }
 
-template<class T> 
-void CaloTruthAccumulator::accumulateEvent( const T& event, 
-					    const edm::EventSetup& setup, 
-					    const edm::Handle< edm::HepMCProduct >& hepMCproduct) {
-  //
-  // Get the collections
-  //
-  
-  edm::Handle< std::vector<reco::GenParticle> > hGenParticles;
-  edm::Handle< std::vector<int> > hGenParticleIndices;
-  
-  event.getByLabel( simTrackLabel_, hSimTracks );
-  event.getByLabel( simVertexLabel_, hSimVertices );
-  
-  event.getByLabel( genParticleLabel_, hGenParticles );
-  event.getByLabel( genParticleLabel_, hGenParticleIndices );
-    
-  std::vector<std::pair<DetId,const PCaloHit*> > simHitPointers;
-  fillSimHits( simHitPointers, event, setup ); 
-  
+template <class T>
+void CaloTruthAccumulator::accumulateEvent(
+    const T& event, const edm::EventSetup& setup,
+    const edm::Handle<edm::HepMCProduct>& hepMCproduct) {
+
+  edm::Handle<std::vector<reco::GenParticle> > hGenParticles;
+  edm::Handle<std::vector<int> > hGenParticleIndices;
+
+  event.getByLabel(simTrackLabel_, hSimTracks);
```

Comment:
same question about `getByLabel()` vs `getByToken()`

Reply:
That's a valid question. This part of the code was simply c&p from the previous incarnation. I quickly looked at all the derived classes from `DigiAccumulatorMixMod` and all of them use `getByLabel()`. At this point, I'm wondering if there is some hidden reason I'm not aware of, or simply that's the current state of the code. I'd like to better understand this before breaking the symmetry.

Reply:
My (limited) understanding is that the MixingModule accumulators have to use `getByLabel()` (e.g. `DigiAccumulatorMixMod` does not inherit from `EDConsumerBase` so `consumes()` is not available, not sure if `ConsumesCollector` could be somehow percolated to these classes but I'd imagine that to require larger changes to MixingModule infrastructure beyond this PR).

Reply:
Ah, perhaps I've learned something new then. @mdhildreth @Dr15Jones can you clarify about this? (It does look like this class is passed `ConsumesCollector` upon construction...)

Reply:
If I remember right, the accumulators are getting data from the 'mixed' event which does not have the infrastructure we use to support `getByToken`.